Add "how to build/run local version of Auspice CLI" to dev docs

I'm trying to test @jameshadfield's PR #1655. The new feature applies only to very specific datasets, none of which are in the sample datasets that come with the heroku test build.

So I need to run auspice locally and pass in my own dataset through the CLI.

However, being a JS noob, I can't figure out how. And unfortunately the dev_docs.md don't seem to help me neither.

It would be nice if there was a simple section explaining how to run the auspice CLI based off the current code base, the equivalent of `pip install -e .` (which would also make the CLI entry point available in path), or something more like `pipx` not putting it in path but allowing me to pass execute it once, something like that must exist!

I want something like `cargo run` (Rust) where it runs the CLI off the current code base.

Ok, here's one issue, which may be related to yours: the current instructions for developers assume you already have a `node_modules/` bootstrapped somehow.

```
rm -rf node_modules/
conda env remove --name auspice
mamba create --name auspice nodejs=16
conda activate auspice
npm install --global .
```

The last line, because of the `prepare` and `build` aliases,

https://github.com/nextstrain/auspice/blob/471fab1424be784bccf3d4470a944adabaeb3442/package.json#L22-L23

tries to run Auspice before any deps are installed, yielding:

```
npm ERR! code 1
npm ERR! path /home/tom/nextstrain/auspice
npm ERR! command failed
npm ERR! command sh -c -- npm run build
npm ERR! > auspice@2.45.2 build
npm ERR! > node auspice.js build --verbose
npm ERR! node:internal/modules/cjs/loader:998
npm ERR!   throw err;
npm ERR!   ^
npm ERR! 
npm ERR! Error: Cannot find module 'argparse'
npm ERR! Require stack:
npm ERR! - /home/tom/nextstrain/auspice/auspice.js
npm ERR!     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:995:15)
npm ERR!     at Function.Module._load (node:internal/modules/cjs/loader:841:27)
npm ERR!     at Module.require (node:internal/modules/cjs/loader:1067:19)
npm ERR!     at require (node:internal/modules/cjs/helpers:103:18)
npm ERR!     at Object.<anonymous> (/home/tom/nextstrain/auspice/auspice.js:3:18)
npm ERR!     at Module._compile (node:internal/modules/cjs/loader:1165:14)
npm ERR!     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1219:10)
npm ERR!     at Module.load (node:internal/modules/cjs/loader:1043:32)
npm ERR!     at Function.Module._load (node:internal/modules/cjs/loader:878:12)
npm ERR!     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12) {
npm ERR!   code: 'MODULE_NOT_FOUND',
npm ERR!   requireStack: [ '/home/tom/nextstrain/auspice/auspice.js' ]
npm ERR! }

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/tom/.npm/_logs/2023-03-23T22_43_28_039Z-debug-0.log
```

--- a/DEV_DOCS.md
+++ b/DEV_DOCS.md
@@ -19,9 +19,24 @@ This project strictly adheres to the [Contributor Covenant Code of Conduct](http
 
 Please see the [project boards](https://github.com/orgs/nextstrain/projects) for currently available issues.
 
-## Contributing code
+## Developer Installation
 
-Code contributions are welcomed! [Please see the main auspice docs](https://docs.nextstrain.org/projects/auspice/en/latest/introduction/install.html) for details on how to install and run auspice from source.
+This is useful for debugging, modifying the source code, or using an unpublished feature branch.
+
+You can install using your system `npm` or [within a Conda environment](https://docs.nextstrain.org/projects/auspice/en/stable/introduction/install.html#install-dependencies) to keep things isolated.
+
+```sh
+# grab the GitHub auspice repo
+git clone https://github.com/nextstrain/auspice.git
+cd auspice
+
+# install dependencies and make `auspice` available globally
+npm install --global .
+```
+
+Updating Auspice should only require pulling the new version from GitHub - it shouldn't require any `npm` commands. You will, however, have to re-build Auspice whenever the client-related code has changed, via `auspice build`.
+
+## Contributing code
 
 Please comment on an open issue if you are working on it.
 For changes unrelated to an open issue, please make an issue outlining what you would like to change/add.

--- a/DEV_DOCS.md
+++ b/DEV_DOCS.md
@@ -30,7 +30,10 @@ You can install using your system `npm` or [within a Conda environment](https://
 git clone https://github.com/nextstrain/auspice.git
 cd auspice
 
-# install dependencies and make `auspice` available globally
+# install dependencies and build auspice
+npm ci
+
+# make `auspice` available globally
 npm install --global .
 ```
 

--- a/docs/introduction/install.rst
+++ b/docs/introduction/install.rst
@@ -36,18 +36,7 @@ If you look at the :doc:`release notes <../releases/changelog>` you can see the 
 Install Auspice as a developer
 ==============================
 
-This is useful for debugging, modifying the source code, or using an unpublished feature branch.
-
-.. code:: bash
-
-   # grab the GitHub auspice repo
-   git clone https://github.com/nextstrain/auspice.git
-   cd auspice
-
-   # install dependencies and make `auspice` available globally
-   npm install --global .
-
-Updating Auspice should only require pulling the new version from GitHub - it shouldn't require any ``npm`` commands. You will, however, have to re-build Auspice whenever the client-related code has changed, via ``auspice build``.
+See `DEV_DOCS.md <https://github.com/nextstrain/auspice/blob/-/DEV_DOCS.md#developer-installation>__`.
 
 Testing if it worked
 ====================

